src/main/java/com/example/javacohort3/ZipCodeBank/services/BillService.java

```diff
@@ -0,0 +1,61 @@
+package com.example.javacohort3.ZipCodeBank.services;
+
+import com.example.javacohort3.ZipCodeBank.exceptions.ResourceNotFoundException;
+import com.example.javacohort3.ZipCodeBank.repositories.AccountRepository;
+import com.example.javacohort3.ZipCodeBank.repositories.CustomerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class BillService {
+    private BillRepository billRepository;
+    private AccountRepository accountRepository;
+    private CustomerRepository customerRepository;
+
+
+    @Autowired
+    public BillService(BillRepository billRepository, AccountRepository accountRepository){
+        this.billRepository = billRepository;
+        this.accountRepository = accountRepository;
+    }
+
+    public void verifyAccountById(Long id){
+        if(accountRepository.findAccountById(id) == null) throw new ResourceNotFoundException();
+    }
+
+    public void verifyBillById(Long id){
+        if(billRepository.findBillById(id) == null) throw new ResourceNotFoundException();
+    }
+
+    public Bill createBill(Bill bill){
+        return billRepository.save(bill);
+    }
+
+    public Bill getBillById(Long id){
+        return billRepository.findBillById(id);
+    }
+
+    public List<Bill> getAllBillsByAccountId(Long accountId){
+        verifyAccountById(accountId);
+
+        ArrayList<Bill> bills = (ArrayList<Bill>) billRepository.findAll();
+        bills.removeIf(a -> !a.getAccount().getId().equals(accountId));
```

Comment:
do you know what this line is doing?

Reply:
Do you know why this works

Reply:
first of did either of you look at the second commit